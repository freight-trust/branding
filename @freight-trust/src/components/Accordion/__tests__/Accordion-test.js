/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { render, cleanup } from '@carbon/test-utils/react';
import { mount } from 'enzyme';
import React from 'react';
import { default as Accordion, AccordionItem } from '../';

describe('Accordion', () => {
  afterEach(cleanup);

  it('should render', () => {
    const wrapper = mount(
      <Accordion className="extra-class">
        <AccordionItem className="child" title="Heading A">
          Panel A
        </AccordionItem>
        <AccordionItem className="child" title="Heading B">
          Panel B
        </AccordionItem>
        <AccordionItem className="child" title="Heading C">
          Panel C
        </AccordionItem>
      </Accordion>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should have no AVT1 violations', async () => {
    render(
      <Accordion className="extra-class">
        <AccordionItem className="child" title="Heading A">
          Panel A
        </AccordionItem>
        <AccordionItem className="child" title="Heading B">
          Panel B
        </AccordionItem>
        <AccordionItem className="child" title="Heading C">
          Panel C
        </AccordionItem>
      </Accordion>
    );

    await expect(document).toHaveNoAxeViolations();
  });
});
