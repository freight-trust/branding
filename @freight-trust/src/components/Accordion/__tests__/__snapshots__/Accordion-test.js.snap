// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render 1`] = `
<Accordion
  align="end"
  className="extra-class"
>
  <ul
    className="bx--accordion extra-class bx--accordion--end"
  >
    <AccordionItem
      className="child"
      title="Heading A"
    >
      <li
        className="bx--accordion__item child"
        onAnimationEnd={[Function]}
      >
        <defaultRenderExpando
          aria-controls="accordion-item-1"
          aria-expanded={false}
          className="bx--accordion__heading"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <button
            aria-controls="accordion-item-1"
            aria-expanded={false}
            className="bx--accordion__heading"
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <ForwardRef(ChevronRight16)
              className="bx--accordion__arrow"
            >
              <Icon
                className="bx--accordion__arrow"
                fill="currentColor"
                height={16}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <svg
                  aria-hidden={true}
                  className="bx--accordion__arrow"
                  fill="currentColor"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
              </Icon>
            </ForwardRef(ChevronRight16)>
            <div
              className="bx--accordion__title"
            >
              Heading A
            </div>
          </button>
        </defaultRenderExpando>
        <div
          className="bx--accordion__content"
          id="accordion-item-1"
        >
          Panel A
        </div>
      </li>
    </AccordionItem>
    <AccordionItem
      className="child"
      title="Heading B"
    >
      <li
        className="bx--accordion__item child"
        onAnimationEnd={[Function]}
      >
        <defaultRenderExpando
          aria-controls="accordion-item-2"
          aria-expanded={false}
          className="bx--accordion__heading"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <button
            aria-controls="accordion-item-2"
            aria-expanded={false}
            className="bx--accordion__heading"
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <ForwardRef(ChevronRight16)
              className="bx--accordion__arrow"
            >
              <Icon
                className="bx--accordion__arrow"
                fill="currentColor"
                height={16}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <svg
                  aria-hidden={true}
                  className="bx--accordion__arrow"
                  fill="currentColor"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
              </Icon>
            </ForwardRef(ChevronRight16)>
            <div
              className="bx--accordion__title"
            >
              Heading B
            </div>
          </button>
        </defaultRenderExpando>
        <div
          className="bx--accordion__content"
          id="accordion-item-2"
        >
          Panel B
        </div>
      </li>
    </AccordionItem>
    <AccordionItem
      className="child"
      title="Heading C"
    >
      <li
        className="bx--accordion__item child"
        onAnimationEnd={[Function]}
      >
        <defaultRenderExpando
          aria-controls="accordion-item-3"
          aria-expanded={false}
          className="bx--accordion__heading"
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <button
            aria-controls="accordion-item-3"
            aria-expanded={false}
            className="bx--accordion__heading"
            onClick={[Function]}
            onKeyDown={[Function]}
            type="button"
          >
            <ForwardRef(ChevronRight16)
              className="bx--accordion__arrow"
            >
              <Icon
                className="bx--accordion__arrow"
                fill="currentColor"
                height={16}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <svg
                  aria-hidden={true}
                  className="bx--accordion__arrow"
                  fill="currentColor"
                  focusable="false"
                  height={16}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
              </Icon>
            </ForwardRef(ChevronRight16)>
            <div
              className="bx--accordion__title"
            >
              Heading C
            </div>
          </button>
        </defaultRenderExpando>
        <div
          className="bx--accordion__content"
          id="accordion-item-3"
        >
          Panel C
        </div>
      </li>
    </AccordionItem>
  </ul>
</Accordion>
`;
